### PR TITLE
Restore ECR repo

### PR DESCRIPTION
**Pull request by ben-harvey**
The MAC-FC team archived the GitHub repo `mac-fc-infra`, which previously managed the ECR repo for Security Hub Collector images. Those resources are now managed in this GitHub repo.

Changes include: 

- an `init` folder that sets up a remote backend in the MACBIS Shared DSO dev account. Local state for these resources is kept in this repo
- an `account` folder that includes the ECR repo and GitHub OIDC resources
  - other AWS accounts that want to pull the Collector image are now whitelisted (rather than the previous approach granting `*` access. This has the benefit of keeping a list of the accounts of current product team users stored in our repo, though we should check in with these teams periodically and remove their access if they no longer run the Collector)
- refactoring of the `build-and-deploy` workflows to:
  - remove the workflow that deployed to the test account (which has been deleted)
  - use GitHub OIDC instead of access key secrets (I deleted the secrets via the GitHub UI) 
  - simplify the tags applied to the Collector image
    - keep the `v2` tag. Please see https://github.com/Enterprise-CMCS/mac-fc-security-hub-collector/pull/23 for context on `v2` vs the `latest` image tag. After this change is merged we can update the GitHub version tag to `v2` to match the image tag

All Terraform in this change has been applied.

Testing:
- tested cross-account pull policy by:
  - temporarily adding an account we control to the whitelist and pulling the image
  -  having MACPro DSO pull the image and verify that it runs the Collector successfully
  - trying to pull from a non-whitelisted account and failing as expected
- tested workflow changes by: 
  - temporarily updating the workflow trigger and building and pushing successfully with GitHub OIDC

### CODE REVIEW

**Comment by leslie-corbalt** [comment]
> Terraform for the ECR repo that hosts the Collector images was previously maintained in the `mac-fc-infra` repo. We're deprecating that repo, so those resources are moving into management by this repo. This change includes:
> 
> * an `init` folder that sets up a remote backend in the MACBIS Shared DSO dev account. Local state for these resources is kept in this repo
> * an `account` folder that includes the ECR repo and GitHub OIDC resources
>   
>   * other AWS accounts that want to pull the Collector image are now whitelisted (rather than the previous approach granting `*` access. This has the benefit of keeping a list of the accounts of current product team users stored in our repo, though we should check in with these teams periodically and remove their access if they no longer run the Collector)
> * refactoring of the `build-and-deploy` workflows to:
>   
>   * remove the workflow that deployed to the test account (which has been deleted)
>   * use GitHub OIDC instead of access key secrets (I deleted the secrets via the GitHub UI)
>   * simplify the tags applied to the Collector image
>     
>     * keep the `v2` tag. Please see [Add region and environment to output #23](https://github.com/Enterprise-CMCS/mac-fc-security-hub-collector/pull/23) for context on `v2` vs the `latest` image tag. After this change is merged we can update the GitHub version tag to `v2` to match the image tag
> 
> All Terraform in this change has been applied.
> 
> Testing:
> 
> * tested cross-account pull policy by:
>   
>   * temporarily adding an account we control to the whitelist and pulling the image
>   * having MACPro DSO pull the image and verify that it runs the Collector successfully
>   * trying to pull from a non-whitelisted account and failing as expected
> * tested workflow changes by:
>   
>   * temporarily updating the workflow trigger and building and pushing successfully with GitHub OIDC

Some comments:
replace `We're deprecating that repo` with `The MAC-FC team archived the MAC Shared DSO Test account and moved those resources into this repo.`.

**Comment by leslie-corbalt** [comment]
> Terraform for the ECR repo that hosts the Collector images was previously maintained in the `mac-fc-infra` repo. We're deprecating that repo, so those resources are moving into management by this repo. This change includes:
> 
> * an `init` folder that sets up a remote backend in the MACBIS Shared DSO dev account. Local state for these resources is kept in this repo
> * an `account` folder that includes the ECR repo and GitHub OIDC resources
>   
>   * other AWS accounts that want to pull the Collector image are now whitelisted (rather than the previous approach granting `*` access. This has the benefit of keeping a list of the accounts of current product team users stored in our repo, though we should check in with these teams periodically and remove their access if they no longer run the Collector)
> * refactoring of the `build-and-deploy` workflows to:
>   
>   * remove the workflow that deployed to the test account (which has been deleted)
>   * use GitHub OIDC instead of access key secrets (I deleted the secrets via the GitHub UI)
>   * simplify the tags applied to the Collector image
>     
>     * keep the `v2` tag. Please see [Add region and environment to output #23](https://github.com/Enterprise-CMCS/mac-fc-security-hub-collector/pull/23) for context on `v2` vs the `latest` image tag. After this change is merged we can update the GitHub version tag to `v2` to match the image tag
> 
> All Terraform in this change has been applied.
> 
> Testing:
> 
> * tested cross-account pull policy by:
>   
>   * temporarily adding an account we control to the whitelist and pulling the image
>   * having MACPro DSO pull the image and verify that it runs the Collector successfully
>   * trying to pull from a non-whitelisted account and failing as expected
> * tested workflow changes by:
>   
>   * temporarily updating the workflow trigger and building and pushing successfully with GitHub OIDC

These sentences can be more precise:
`Terraform for the ECR repo that hosts the Collector images was previously maintained in the `mac-fc-infra` repo. We're deprecating that repo, so those resources are moving into management by this repo.`
A suggestion:
`The MAC-FC team archived the GitHub repo mac-fc-infra, which managed the ECR repo for Collector images. Those resources are now managed in this GitHub repo.`

File: README.md [hunk 1]
```diff
@@ -67,7 +67,7 @@ Example usage:
 ```terraform
   module "security_hub_collector_ecr" {
     source = "github.com/CMSgov/security-hub-collector"
```
[the remark this answers is not in the document]
Comment (leslie-corbalt) [reply README.md]: The Installation section of the README is out of date.

**Review by leslie-corbalt — COMMENTED** [review]
1. I had a comment on the PR description.
2. There is a `lifecycle-policy.json` at the top-level of the repo, which is no longer used since you created it again in the ecr_lifecycle_policy resource
3. Did you update the lockfile for all target platforms (OS, hardware)? (`terraform providers lock -platform={linux,darwin}_{amd64,arm64}`?

File: terraform/dev/account/github-oidc.tf [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    ]
+    resources = ["arn:aws:ecr:us-east-1:${data.aws_caller_identity.current.account_id}:repository/security-hub-collector"]
+  }
+}
+
+resource "aws_iam_policy" "github_actions_permissions" {
+  name        = "github-actions-permissions"
+  policy      = data.aws_iam_policy_document.github_actions_permissions.json
+  path        = "/delegatedadmin/developer/"
+  description = "Permissions for GitHub Actions OIDC"
```
[the remark this answers is not in the document]
Comment (ben-harvey) [reply terraform/dev/account/github-oidc.tf]: I updated the resource names to reflect this, and moved and applied Terraform state

File: terraform/dev/account/variables.tf [hunk 3]
```diff
@@ -1,22 +1,4 @@
...
-
-variable "scan_on_push" {
-  type        = bool
-  description = "Scan image on push to repo."
-  default     = true
-}
-
-variable "allowed_read_principals" {
+variable "ecr_read_aws_accounts" {
   type        = list(any)
```
[the remark this answers is not in the document]
Comment (ben-harvey) [reply terraform/dev/account/variables.tf]: :thumbsup:

File: terraform/dev/init/main.tf [hunk 4]
```diff
@@ -0,0 +1,80 @@
+resource "aws_dynamodb_table" "lock_table" {
+  name                        = "security-hub-collector-dev-lock-table"
+  read_capacity               = 20
+  write_capacity              = 20
+  hash_key                    = "LockID"
+  deletion_protection_enabled = true
+
```
[the remark this answers is not in the document]
Comment (ben-harvey) [reply terraform/dev/init/main.tf:10]: :thumbsup:

**Review by ben-harvey — COMMENTED** [review]
I made the suggested updates to the PR description

**Comment by ben-harvey** [comment]
I re-ran the build-and-push job to test the Terraform changes. All Terraform is applied.

**Review by leslie-corbalt — COMMENTED** [review]
add `*.tfstate.backup` to .gitignore

otherwise, looks good!

**Review by leslie-corbalt — APPROVED** [review]
LGTM!

**Pull request merged**